### ISSUE TITLE
Docs: add minimal examples of a full optimization & CachingOptimizer

### ISSUE THREAD

**Issue by matbesancon** [jump-dev/MathOptInterface.jl#591]
There should be some examples in the documentation on the use of `MOIU.CachingOptimizer`, without that it is hard to get an entry point into MOI (make something work without JuMP).

Cbc for instance is not usable without `MOIU.CachingOptimizer` or JuMP.

**Comment by matbesancon** [comment]
To complement on this issue, errors appearing seem inconsistent with the docstrings and error messages:
```julia
julia> o = SCS.Optimizer()
julia> z = MOI.add_variables(o, 3)
ERROR: MathOptInterface.AddVariableNotAllowed: Adding variables cannot be performed. You may want to use a `CachingOptimizer` in `AUTOMATIC` mode or you may need to call `reset_optimizer` before doing this operation if the `CachingOptimizer` is in `MANUAL` mode.
julia> model = MOI.Utilities.CachingOptimizer(o, MOI.Utilities.AUTOMATIC)
MOIU.CachingOptimizer{MOI.AbstractOptimizer,SCS.Optimizer}
in state NO_OPTIMIZER
in mode AUTOMATIC
with model cache SCS.Optimizer
with optimizer nothing
```

The last command is the first working method that comes to mind following the thrown error but is likely wrong: a user seeing an error because the optimizer does not support incremental changes will likely want to build a caching optimizer with whatever default works as a cache. There is no indication of what object is expected as model cache (if the user does not have another Optimizer type at hand that supports it.

**Comment by matbesancon** [comment]
UniversalFallback should be mentioned or even used as cache in a default method

**Comment by odow** [comment]
Closed by #1272
